src/Surfnet/StepupGateway/GatewayBundle/Sso2fa/DateTime/ExpirationHelper.php

```diff
@@ -0,0 +1,86 @@
+<?php declare(strict_types=1);
+
+/**
+ * Copyright 2023 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupGateway\GatewayBundle\Sso2fa\DateTime;
+
+use DateTime as CoreDateTime;
+use Surfnet\StepupBundle\DateTime\DateTime;
+use Surfnet\StepupGateway\GatewayBundle\Sso2fa\Exception\InvalidAuthenticationTimeException;
+use Surfnet\StepupGateway\GatewayBundle\Sso2fa\ValueObject\CookieValueInterface;
+use TypeError;
+
+class ExpirationHelper implements ExpirationHelperInterface
+{
+    /**
+     * @var CoreDateTime
+     */
+    private $now;
+
+    /*
+     * The period in seconds that we still acknowledge the
+     * cookie even tho the expiration was reached. This accounts
+     * for server time/sync differences that may occur.
+     */
+    private $gracePeriod;
+
+    /*
+     * The SSO on 2FA cookie lifetime in seconds
+     *
+     * See: config/legacy/parameters.yaml sso_cookie_lifetime
+     */
+    private $cookieLifetime;
+
+    public function __construct(int $cookieLifetime, int $gracePeriod, CoreDateTime $now = null)
```

Comment:
Waarom een alias CoreDateTime ipv gewoon DateTime?

Reply:
I tried my best to use the domain specific DateTime implementation. Which produces a /DateTime, which I aliassed to CoreDateTime as our linting discourages using unimported classes.